src/R/Support/Impl/Help/Packages/PackageInfo.cs

```diff
@@ -80,7 +80,7 @@ internal sealed class PackageInfo : NamedItemInfo, IPackageInfo {
                 try {
                     var result = await _host.Session.PackagesFunctionsNamesAsync(Name, REvaluationKind.BaseEnv);
                     functions = result.Children<JValue>().Select(v => (string)v.Value);
-                } catch (TaskCanceledException) { } catch (REvaluationException) { }
+                } catch (RException) { }
```

Comment:
`AD7Property.SetValueAsStringWithError` - executed inside `ThreadHelper.JoinableTaskFactory`, which handles task cancellation
`GridDataSource` doesn't catch `RHostDisconnectedException` because calling code relies on it being thrown so that awaiter continuation doesn't execute.
`REnvironmentProvider` catches both

Generally speaking, I'd rather not change anything about handling `RHostDisconnectedException` because:
- for async code, `OperationCanceledException` is a normal thing and doesn't lead to a crash. 
- `RHostDisconnectedException` may be handled up in the stack separately from the `RException` or intentionally not handled, so we can easily break existing functionality
- I don't remember seeing VS crashed because of `RHostDisconnectedException`